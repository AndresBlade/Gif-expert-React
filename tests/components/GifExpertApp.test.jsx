import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import { GifExpertApp } from "../../src/GifExpertApp"
import UserEvent from '@testing-library/user-event'



describe('Pruebas en <GifExpertApp />', () => {

    jest.setTimeout(10000)

    const category = 'Prueba'

    test('Deberia de hacer match con el snapshot', async () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot()
    })

    test('Debería añadir una nueva categoria', async () => {
        const user = UserEvent.setup()
        render(<GifExpertApp />);
        const input = screen.getByRole('textbox')
        const form = screen.getByRole('form')
        await user.type(input, 'Water')



        fireEvent.submit(form)

        await waitFor(() => {
            expect(screen.getAllByRole('img').length).toBeGreaterThan(0)
        }, { timeout: 5000 })



    })

    test('No debería añadir una nueva categoría si ya existe', async () => {
        const user = UserEvent.setup()
        const { container } = render(<GifExpertApp />);
        const input = screen.getByRole('textbox')
        const form = screen.getByRole('form')

        await user.type(input, 'Water')
        fireEvent.submit(form)

        //Ingresa la misma categoría

        await user.type(input, 'Water')
        fireEvent.submit(form)

        await waitFor(() => {
            expect(container.querySelectorAll('.card-grid')).toHaveLength(1) //Que solo se añada una categoría
        }, { timeout: 5000 })
    })
})