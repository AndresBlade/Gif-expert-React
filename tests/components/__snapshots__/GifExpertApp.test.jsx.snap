// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Deberia de hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    name="gif-search"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
